### PR TITLE
ci: update Cypress to 10

### PR DESCRIPTION
**Pull request by NickDub**
### Pull Request check-list

To ensure your Pull Request can be accepted as fast as possible, make sure to review and check all of these items:

- [ ] If your changes affects code, did your write the tests?
- [ ] Are tests passing? (`npm test` on both front/server)
- [ ] Is the linter passing? (`npm run eslint` on both front/server)
- [ ] Did you run prettier? (`npm run prettier` on both front/server)
- [ ] If you are adding a new features/services, did you run integration comparator? (`npm run compare-translations` on front)
- [ ] Did you test this pull request in real life? With real devices? If this development is a big feature or a new service, we recommend that you provide a Docker image to [the community](https://community.gladysassistant.com/) for testing before merging.
- [ ] If your changes modify the API (REST or Node.js), did you modify the API documentation? (Documentation is based on comments in code)
- [ ] If you are adding a new features/services which needs explanation, did you modify the user documentation? See [the GitHub repo](https://github.com/GladysAssistant/v4-website) and the [website](https://gladysassistant.com).
- [ ] Did you add fake requests data for the demo mode (`front/src/config/demo.js`) so that the demo website is working without a backend? (if needed) See [https://demo.gladysassistant.com](https://demo.gladysassistant.com).

NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open.

### Description of change

- Update some Github actions to prevent this warning: 
`Node.js 12 actions are deprecated. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/. Please update the following actions to use Node.js 16: actions/checkout, actions/setup-node, actions/cache, ...`.
- Refactor NPM cache use.
- Add `.nvmrc` file to use the same Node version between developpers and CI/CD.

### CODE REVIEW

File: .github/workflows/docker-pr-build.yml [hunk 1]
```diff
@@ -134,7 +119,7 @@ jobs:
         run: |
           npm ci && npm run install-server:dev
       - name: ✅ Cypress run
-        uses: cypress-io/github-action@v2
+        uses: cypress-io/github-action@v4
```
Comment (atrovato) [inline .github/workflows/docker-pr-build.yml]: Cannot find cypress!
See logs :)

Comment (NickDub) [reply .github/workflows/docker-pr-build.yml]: Fixed 😉

Comment (NickDub) [reply .github/workflows/docker-pr-build.yml]: Everything is ok except `integration/bluetooth/scanner/BluetoothStateChange.cy.js` which randomly fails...

Comment (NickDub) [reply .github/workflows/docker-pr-build.yml]: @atrovato can you check again ?

Comment (atrovato) [reply .github/workflows/docker-pr-build.yml]: I will. I already "fixed" it many times... but it is still random :'(

**Comment by Pierre-Gilles** [comment]
Thanks for your work @NickDub ! I'm integrating your work into: https://github.com/GladysAssistant/Gladys/pull/1656

**Review by VonOx — APPROVED** [review]

**Comment by Pierre-Gilles** [comment]
Thanks for the PR! I did a few changes here https://github.com/GladysAssistant/Gladys/pull/1661 and merged it on master :)

**Pull request closed without merging**